### ISSUE TITLE
[engine] authorizedRoles does not work as expected

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#4658]
## Describe the bug
When `authorizedRoles` is setup, using targeting for pages in the website editorial bp, when a page has been set to be accessible to `ROLE_user`, and targeting role is set to `ROLE_user`, the page appears in it's category landing page and in site search but clicking on the page from either or the Sidebar it says user does not have enough rights to access the page.  When targeting role is set to `user` (no prefix), the page does not appear in category landing page, does not appear in site search, but you are able to access page from Sidebar

## To Reproduce
1. Using the website editorial bp, setup targeting to use roles **Site Config** -> **Configuration** -> **Targeting Configuration**:
```
<targeting>
   ...
   <property>
        <name>roles</name>
        <label>Role</label>
        <description>User role, e.g. ROLE_admin</description>
        <type>input</type> <!-- valid types: dropdown, checkboxes, input -->
        <default_value></default_value>
        <hint>Setting the role will change content available based on role selected.</hint>
   </property>
</targeting>
```
2. Model criteria in content.  
    - Click on **Site Config** -> **Content Types** then **Open Existing Type** and select **Page - Article**.  
    - Select the `Repeating Group` control and add it to the `Metadata` Form Section.  In the **Repeating Group** properties, set the Title field to “Authorized Roles” and the Name / Variable Name field to “authorizedRoles.”
    - Add an `Input` control inside the `Repeating Group`, with the Title field set to “Role” and the Name / Variable Name field set to “role”. Make this Input required by checking the checkbox under Constraints in the Required field in the Properties Explorer.
    -  Save your changes
    
<img width="605" alt="Screen Shot 2021-04-19 at 11 20 50 AM" src="https://user-images.githubusercontent.com/25483966/115261083-53543980-a101-11eb-80b2-57edef1eff2b.png">

3. Update content.  Edit the page `Top Books For Young Women`.  Scroll down to `Authorized Roles` in the `Metadata` section, then click on `Add First Item`
![tagging-roles-tag-content-1](https://user-images.githubusercontent.com/25483966/115261747-e2615180-a101-11eb-8cff-3b41a9e610c2.png)
Input `ROLE_user` in the `Role` field
![tagging-roles-tag-content-2](https://user-images.githubusercontent.com/25483966/115261848-fb6a0280-a101-11eb-9f35-9806f8d71db4.png)
Save your changes
4. Click on the `Targeting` icon at the top right and set the `Role` to `ROLE_user`
![tagging-roles-targeting](https://user-images.githubusercontent.com/25483966/115262048-2b190a80-a102-11eb-88e5-8284e65cbdf3.png)
5. Preview category landing page `Entertainment`.  Notice that the page `Top Books For Young Women` is listed.
6. Click on the page `Top Books For Young Women` , notice the message `The user doesn't have enough rights to access the page.` appears
<img width="1202" alt="Screen Shot 2021-04-19 at 11 28 23 AM" src="https://user-images.githubusercontent.com/25483966/115262358-77644a80-a102-11eb-9117-11bb1b014af7.png">
7. Do a site search by clicking on the hamburger icon  then enter `books` for the search term.  Notice that the article `Top Books For Young Women` is listed.  Click on the search result, you'll get the message `The user doesn't have enough rights to access the page.`
8. Edit the page `Top Books For Young Women` and change the value in `Role` under `Authorized Roles` to `user`
9. Preview category landing page `Entertainment`.  Notice that the page `Top Books For Young Women` is not listed.
10. Do a site search for `books`.  Notice that the page `Top Books For Young Women` is not listed.
11. Open the Sidebar and click on `Top Books For Young Women` and notice that you are able to preview the article

## Expected behavior
When `authorizedRoles` and targeting is setup for pages, the page should appear in it's category landing page and site search and the user should be able to preview the page from either category landing page, site search or the Sidebar

## Screenshots
{{If applicable, add screenshots to help explain your problem.}}

## Logs
{{If applicable, attach the logs/stack trace (use https://gist.github.com).}}

### Version
Studio Version Number: 3.1.14-SNAPSHOT-91da16
Build Number: 91da163a2cbfacf6624b1b6d4b9a978a339f9add
Build Date/Time: 04-19-2021 11:39:44 -0400

### OS
macOS 10.15.7

### Browser
Chrome browser

## Additional context
{{Add any other context about the problem here.}}

**Comment by alhambrav** [comment]
Verified fixed in 3.1

**Comment by alhambrav** [comment]
Verified fixed in 4.0
